### PR TITLE
doc: add project structure file

### PR DESCRIPTION
**Pull request by kwy95**
Adds a documentation file describing the overall structure of kw.

Signed-off-by: Rubens Gomes Neto <rubens.gomes.neto@alumni.usp.br>

### CODE REVIEW

File: documentation/content/project_structure.rst [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  {
+    ...
+  }
+
+This standardized structure makes the components easier to read and understand,
+as the top of the file is dedicated to explain the component, and harbors the
+entry point function that organizes the component's function. Furthermore, the
+options parser and help functions are located at the bottom of the file,
+allowing for the middle section to be dedicated to the actual implementation of
+the components functionality. If you understand this organization, you can
```
Comment (mairacanal) [inline documentation/content/project_structure.rst:82]:
```suggestion
the components' functionality. If you understand this organization, you can
```

Just a small grammar pick.

**Review by mairacanal — COMMENTED** [review]
Great text! As a beginner in kw development, it was very useful to understand more about the project structure.

File: documentation/content/project_structure.rst [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+avoid common Bash mistakes and `shfmt <https://github.com/mvdan/sh>`_ that
+enforces code-style patterns. Other rules are described in the :ref:`code-style
+<coding-style>` documentation. In this sense, this page describes the overall
+code architecture and some details associated with the code.
+
+If you are starting to work with **kw**'s source, you should see it in five
+different levels: hub (`kw` file), components, libraries, plugins, and
+documentation. Some of it is shown in the picture below.
+
+.. image:: project_structure.png
```
Comment (rodrigosiqueira) [inline documentation/content/project_structure.rst:24]: let's use the svg file

**Comment by kwy95** [comment]
@mairacanal @rodrigosiqueira Thanks for the reviews, I addressed all of your concerns and added a new section explaining our documentation.

**Comment by rodrigosiqueira** [comment]
Nice PR! Merged to the unstable branch.

**Pull request closed without merging**